osquery/tables/system/windows/users.cpp

```diff
@@ -110,15 +125,45 @@ void processRoamingProfiles(const std::set<std::string>& processedSids,
     ret = LookupAccountSidW(
         nullptr, sid, accntName, &accntNameLen, domName, &domNameLen, &eUse);
     r["username"] = ret != 0 ? wstringToString(accntName) : "";
-    LocalFree(sid);
 
     results.push_back(r);
   }
 }
 
-void processLocalAccounts(std::set<std::string>& processedSids,
+// Enumerate the users from the profiles key in the Registry, matching only
+// the UIDs/RIDs (if any) and skipping any SIDs of local-only users that
+// were already processed in the earlier API-based enumeration.
+void processRoamingProfiles(const std::set<std::string>& selectedUids,
+                            const std::set<std::string>& processedSids,
+                            QueryData& results) {
+  QueryData regResults;
+  queryKey(kRegProfilePath, regResults);
+
+  for (const auto& profile : regResults) {
+    Row r;
+    if (profile.at("type") != "subkey") {
+      continue;
+    }
+
+    auto sidString = profile.at("name");
+
+    if (sidMatchesAnyDesiredUids(selectedUids, sidString)) {
+      // Skip this user if already processed
+      if (processedSids.find(sidString) == processedSids.end()) {
+        genUser(sidString, results);
+      }
+    }
+  }
+}
+
+// Enumerate all local users, constraining results to the list of UIDs if
+// any, and recording all enumerated users' SIDs to exclude later from the
+// walk of the Roaming Profiles key in the registry.
+void processLocalAccounts(const std::set<std::string>& selectedUids,
+                          std::set<std::string>& processedSids,
                           QueryData& results) {
-  unsigned long dwUserInfoLevel = 3;
+  unsigned long dwUserInfoLevel = 0; // retrieve username with NetUserEnum()
```

Comment:
nit: I _definitely_ could be wrong as I've been out of the loop for a while, but I thought we preferred keeping comments on their own line, as opposed to on a code line?

Reply:
Trailing comments are implicitly allowed by clang-format and our clang-format config specifies their spacing; actually clang-format doesn't appear to have a way to enforce their use or not.

Is there a style guide that discourages them? Here they're short and stay to the 80-character column limit.

But, I made the requested change here.